data/list/basic.lean

```diff
@@ -1160,6 +1160,22 @@ by induction l; simp [list.join, *] at *
 
 end monoid
 
+@[simp, to_additive list.sum_erase]
+theorem prod_erase [decidable_eq α] [comm_monoid α] {a} : 
+  Π {l : list α}, a ∈ l → a * (l.erase a).prod = l.prod
+| []          := λ h, absurd h $ not_mem_nil _
```

Comment:
instead of `λ h,` move the `h` to the list left of `:=`

Reply:
Sure, I just did the corresponding changes

Reply:
I did a couple of other changes to, especially introducing a special split rule for `a ∈ l` so the simplifier does most of the job now.

Reply:
Awesome. I'm a bit surprised that the simplifier could figure it out.

Reply:
After the `rcases eq_or_ne_mem_of_mem ...` its straight forwar rewriting! So it works nicely :-)